nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -598,23 +598,39 @@ sub multiLineList {
 ${\join "", (map { "  $_\n" } (uniq @attrs))}}
 EOF
 
-sub generateNetworkingDhcpConfig {
-    # FIXME disable networking.useDHCP by default when switching to networkd.
+sub generateNetworkingConfig {
     my $config = <<EOF;
-  # Enables DHCP on each ethernet and wireless interface. In case of scripted networking
-  # (the default) this is the recommended approach. When using systemd-networkd it's
-  # still possible to use this option, but it's recommended to use it in conjunction
-  # with explicit per-interface declarations with `networking.interfaces.<interface>.useDHCP`.
-  networking.useDHCP = lib.mkDefault true;
+  # Use systemd-networkd to enable DHCP (IPv4) and Router
+  # Advertisements (IPv6) on every ethernet and wireless interface.
+  systemd.network = {
+    enable = true;
+    networks = {
+      # Use lower numbers for additional profiles, since networkd
+      # processes these alphanumerically and every interface will only
+      # ever use its first match.
+      "99-autoconfig-ether" = {
+        matchConfig.Type = [
+          "ether"
+        ];
+        networkConfig = {
+          DHCP = "ipv4";
+          IPv6AcceptRA = true;
```

Comment:
`IPv6AcceptRA`  already defaults to being enabled:

> The default is to disable RA reception for bridge devices or when IP forwarding is enabled, and to enable it otherwise. Cannot be enabled on devices aggregated in a bond device or when link-local addressing is disabled.